src/SlidingSyncManager.ts

```diff
@@ -359,10 +359,14 @@ export class SlidingSyncManager {
         let proxyUrl: string | undefined;
 
         try {
-            const clientWellKnown = await AutoDiscovery.findClientConfig(client.getDomain()!);
+            const clientDomain = await client.getDomain();
+            if (clientDomain === null) {
+                throw new RangeError("Homeserver domain is null");
+            }
+            const clientWellKnown = await AutoDiscovery.findClientConfig(clientDomain);
             proxyUrl = clientWellKnown?.["org.matrix.msc3575.proxy"]?.url;
         } catch (e) {
-            // client.getDomain() is invalid, `AutoDiscovery.findClientConfig` has thrown
+            // client.getDomain() is null, we've thrown
```

Comment:
`AutoDiscovery.findClientConfig ` still could have thrown though from it being invalid in a different way, so I think the comment probably needs to have both of these cases.